plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueExpressionUtil.java

```diff
@@ -88,7 +89,14 @@ public static String buildGlueExpression(List<String> columnNames, TupleDomain<S
         List<String> perColumnExpressions = new ArrayList<>();
         int expressionLength = 0;
         Map<String, Domain> domains = partitionKeysFilter.getDomains().get();
-        for (String columnName : columnNames) {
+        // Glue API does not accept the usage of specific column names in the filter expression
+        List<String> validGlueFilterColumnNames = columnNames.stream()
+                .filter(columnName -> !columnName.equalsIgnoreCase("key"))
+                .collect(Collectors.toList());
+        if (validGlueFilterColumnNames.isEmpty()) {
+            return "";
+        }
+        for (String columnName : validGlueFilterColumnNames) {
```

Comment:
```suggestion
        for (String columnName : columnNames) {
            if (columnNam.equalsIgnoreCase("key")) {
                // .... explanation ...
                continue;
            }
```

Reply:
+1 on this recommendation

Reply:
I looked through the grammar of jsqlparser (which is supposedly used internally by AWS Glue for parsing the partition name SQL expression and I discovered that whenever a TOKEN entry (BY, DO, IS, IN, OR, ON, ALL, AND, KEY, ....) is being used, the issue I reported can be reproduced.

https://sourceforge.net/p/jsqlparser/code/HEAD/tree/trunk/src/main/javacc/JSqlParserCC.jj

I have asked in the meantime AWS Support to verify whether there is though a way to escape column names in the filter.